### ISSUE TITLE
TypeError in create_dataset with specified chunks

### ISSUE THREAD

**Issue by schneeammer** [h5py/h5py#900]
First of all I'm really sorry if it's my own mistake, I'm just about to learn using h5py.
I can't set the chunks for multidimensional datasets. The error message looks like it's a typo in the code of dataset.py but maybe I get something wrong here.

```
Traceback (most recent call last):
  File "/Users/jon/Code/grassland/dev/metscr/tests/test_hdf5.py", line 378, in test_vals
    self.df.set_vals('/TEST/VALS/', 'withattrs', val1, colnames=colnames)
  File "/Users/jon/Code/grassland/dev/metscr/tools/hdf5.py", line 501, in set_vals
    maxshape=maxshape, chunks=chunksize)
  File "//anaconda/lib/python3.5/site-packages/h5py/_hl/group.py", line 105, in create_dataset
    dsid = dataset.make_new_dset(self, shape, dtype, data, **kwds)
  File "//anaconda/lib/python3.5/site-packages/h5py/_hl/dataset.py", line 76, in make_new_dset
    if isinstance(chunks, tuple) and (-numpy.array([ i>=j for i,j in zip(tmp_shape,chunks) if i is not None])).any():
TypeError: The numpy boolean negative, the `-` operator, is not supported, use the `~` operator or the logical_not function instead.
```

inside a class that inherits from h5py.File:
```
import numpy as np
...
path_ds = "/path/in/my/hdf5/document"
vals =np.array(...)
...
maxshape = tuple(None if i == 0 else vals.shape[i] for i in range(n))
chunksize = tuple(chk if i == 0 else maxshape[i] for i in range(n))
# Next line rises the Exception, but only if 'chunks' is specified:
self.create_dataset(path_ds, data=vals, maxshape=maxshape, chunks=chunksize)
```

Same error can be reproduced:
```
>>> import numpy as np
>>> -np.array([True, True, False, False])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: The numpy boolean negative, the `-` operator, is not supported, use the `~` operator or the logical_not function instead.
```

### Technical details
MacOS 10.10.5
h5py    2.6.0
HDF5    1.8.17
Python  3.5.2 |Anaconda custom (x86_64)| (default, Jul  2 2016, 17:52:12) 
[GCC 4.2.1 Compatible Apple LLVM 4.2 (clang-425.0.28)]
sys.platform    darwin
sys.maxsize     9223372036854775807
numpy   1.13.0

Thank you for your time!

**Comment by schneeammer** [comment]
Actually looking at the code I realized that it looks like this already have been fixed. sry

**Comment by tacaswell** [comment]
For reference, fixed by 973c47f2fb70294ef0cba37a265360e7603f0528 / https://github.com/h5py/h5py/pull/683 and release in h5py 2.7

@schneeammer Thanks for reporting (even though it was already fixed)!
